python/metric_pipeline_utils/repeatability.py

```diff
@@ -0,0 +1,316 @@
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+import astropy.units as u
+import math
+import numpy as np
+import random
+from scipy.stats import norm
+
+import lsst.pipe.base as pipeBase
+from lsst.verify import Measurement, Datum
+
+thousandDivSqrtTwo = 1000/math.sqrt(2)
+
+
+def measurePhotRepeat(metric, filterName, *args, **kwargs):
+    """Measurement of a photometric repeatability metric across a set of
+    observations.
+
+    Parameters
+    ----------
+    metric : `lsst.verify.Metric`
+        A Metric to construct a Measurement for.
+    filterName : `str`
+        Name of filter used for all observations.
+    *args
+        Additional arguments to pass to `calcPhotRepeat`.
+    **kwargs
+        Additional keyword arguments to pass to `calcPhotRepeat`.
+
+    Returns
+    -------
+    measurement : `lsst.verify.Measurement`
+        Measurement of the repeatability and its associated metadata.
+
+    See also
+    --------
+    calcPhotRepeat: Computes statistics of magnitudes differences of sources across
+        multiple visits. This is the main computation function behind
+        repeatability measurement.
+    """
+    results = calcPhotRepeat(*args, **kwargs)
+    datums = {}
+    datums['filter_name'] = Datum(filterName, label='filter',
+                                  description='Name of filter for this measurement')
+    datums['rms'] = Datum(results['rms'], label='RMS',
+                          description='Photometric repeatability RMS of stellar pairs for '
+                          'each random sampling')
+    datums['iqr'] = Datum(results['iqr'], label='IQR',
+                          description='Photometric repeatability IQR of stellar pairs for '
+                          'each random sample')
+    datums['magDiff'] = Datum(results['magDiff'], label='Delta mag',
+                              description='Photometric repeatability differences magnitudes for '
+                              'stellar pairs for each random sample')
+    datums['magMean'] = Datum(results['magMean'], label='mag',
+                              description='Mean magnitude of pairs of stellar sources matched '
+                              'across visits, for each random sample.')
+    return Measurement(metric, results['repeatability'], extras=datums)
```

Comment:
I'd rather start by not adding extras to `Measurement` objects and only add things in the run method.

Reply:
And actually I'm uncomfortable having this in both the run method and here.  I think I'd avoid porting this over.  If it's here for the docstring, that should probably be attached to the run method in the measurement task instead of here.

Reply:
OK, will think this through more carefully on a different branch.